### ISSUE TITLE
NVM restore: detect incorrectly formatted application version on 800 series

### ISSUE THREAD

**Issue by AlCalzone** [AlCalzone/node-zwave-js#6775]
The 800 series application version NVM file `0x41000` is formatted differently on the Zooz stick than on the firmwares I can flash onto the devkit:

```
Zooz:   00 03 13 07
Devkit: 03 13 07 00
        ┌─ ┌─ ┌─ ┌─
        │  │  │  ??
        │  │  └ MAJOR
        │  └ MINOR
        └ PATCH
```

Since this is stored as little endian, the actual values used in code are `0x07130300` for the Zooz and `0x00071303` for the "official" firmware. When the migration routine for 7.20.0 -> 7.21.0 compares that value with `APP_VERSION_7_21_0` (`0x00071500`), the app version on the Zooz is bigger, so the migration does not trigger, causing the RF settings to be corrupted after a firmware upgrade or restore on a controller running 7.21.0 or higher.

We should detect this situation, fix the application version in NVM, and not rely on the firmware built-in migration in this case.

**Comment by danielsza** [comment]
This seems like the exact issue that I experienced. I purchased a 800 series ethernet adapter https://www.ebay.com/itm/166619186789 I restored my NVM from a Zooz 800 stick and the RF region became corrupt. I was not able to fix the corruption (z/ip, firmware update, NVM restore, reset the NVM) I ended up returning it as a result. 
I believe the seller had to flash it with a tool that I didn't own.

I ended up switching to Razberry Pro 7, along with a Tubesz Z-Wave PoE Kit. I restored the same NVM on this and it didn't corrupt the RF region .
